packages/text-field/index.js

```diff
@@ -0,0 +1,367 @@
+    return classnames('mdc-text-field', Array.from(classList), className, {
+      'mdc-text-field--outlined': outlined,
+      'mdc-text-field--textarea': textarea,
+      'mdc-text-field--fullwidth': fullWidth,
+      'mdc-text-field--disabled': disabled,
+      'mdc-text-field--with-trailing-icon': trailingIcon,
+      'mdc-text-field--with-leading-icon': leadingIcon,
+      'mdc-text-field--box': box,
+      'mdc-text-field--dense': dense,
+    });
+  }
+
+  getIsRtl = () => {
+    if (this.textFieldElement.current) {
+      const dir = window.getComputedStyle(this.textFieldElement.current).getPropertyValue('direction');
+      return dir === 'rtl';
+    }
+  }
+
+  /**
+  * adapter methods
+  */
+
+  get adapter() {
+    const rootAdapterMethods = {
+      addClass: (className) =>
+        this.setState({classList: this.state.classList.add(className)}),
+      removeClass: (className) => {
+        const {classList} = this.state;
+        classList.delete(className);
+        this.setState({classList});
+      },
+      hasClass: (className) => this.classes.split(' ').includes(className),
+      isFocused: () => this.state.isFocused,
+      isRtl: this.getIsRtl,
+    };
+
+    return Object.assign({},
+      rootAdapterMethods,
+      this.inputAdapter,
+      this.labelAdapter,
+      this.lineRippleAdapter,
+      this.notchedOutlineAdapter,
+    );
+  }
+
+  get inputAdapter() {
+    // value: string,
+    // disabled: boolean, --> doesn't need to be implemented since the <INPUT> handles it
+    // also the `get disabled` isn't actually used, except in the vanilla component
+    // validity: {
+    //   badInput: boolean,
+    //   valid: boolean,
+    // },
+
+    return {
+      getNativeInput: () => {
+        const input = {
+          validity: {
+            badInput: this.getBadInput(),
+            valid: this.getIsValid(),
+          },
+        };
+
+        // https://stackoverflow.com/a/44913378
+        Object.defineProperty(input, 'value', {
+          get: () => this.state.value,
+          set: (value) => this.setState({foundationValue: value}),
+        });
+
+        return input;
+      },
+    };
+  }
+
+  get labelAdapter() {
+    return {
+      shakeLabel: (shakeLabel) => {
+        const {floatingLabelElement: floatingLabel} = this;
+        if (shakeLabel && floatingLabel && floatingLabel.current) {
+          floatingLabel.current.shake();
+        }
+      },
+      floatLabel: (shouldFloatLabel) => this.setState({shouldFloatLabel}),
+      hasLabel: () => !!this.props.label,
+      getLabelWidth: () => this.state.labelWidth,
+    };
+  }
+
+  get lineRippleAdapter() {
+    return {
+      activateLineRipple: () => this.setState({activeLineRipple: true}),
+      deactivateLineRipple: () => this.setState({activeLineRipple: false}),
+      setLineRippleTransformOrigin: (lineRippleCenter) => this.setState({rippleCenter: lineRippleCenter}),
```

Comment:
Nit: rename `lineRippleCenter` to `rippleCenter` so you can use the variable-as-key assignment `this.setState({rippleCenter})`